internal/analytics/reporter.go

```diff
@@ -11,4 +12,42 @@ type Reporter interface {
 
 	// RegisterCurrentIdentity loads the current anonymous identity and registers it.
 	RegisterCurrentIdentity(ctx context.Context, k8sCli kubernetes.Interface, cfgDir string) error
+
+	// ReportCommand reports a new executed command. The command should be anonymized before using this method.
+	ReportCommand(platform config.BotPlatform, command string) error
+
+	// ReportBotEnabled reports an enabled bot.
+	ReportBotEnabled(platform config.BotPlatform) error
+
+	// ReportBotHandledEventSuccess reports a successfully handled event using a given bot.
+	ReportBotHandledEventSuccess(platform config.BotPlatform) error
+
+	// ReportBotHandledEventError reports a failure while handling event using a given bot.
+	ReportBotHandledEventError(platform config.BotPlatform, err error) error
+
+	// ReportSinkEnabled reports an enabled sink.
+	ReportSinkEnabled(platform Sink) error
+
+	// ReportSinkHandledEventSuccess reports a successfully handled event using a given sink.
+	ReportSinkHandledEventSuccess(platform Sink) error
+
+	// ReportSinkHandledEventError reports a failure while handling event using a given sink.
+	ReportSinkHandledEventError(platform Sink, err error) error
+
+	// ReportFatalError reports a fatal app error.
+	ReportFatalError(err error) error
+
+	// Close cleans up the reporter resources.
+	Close() error
 }
+
+// Sink defines one-way communication platform.
+type Sink string
```

Comment:
Because it is used just in the reporter. I also thought about that, but I decided to keep it defined where it's actually used. But I can move that to config if that's what you prefer 👍

Reply:
yep, let's aggregate them together for the future usage.